flexget/plugins/input/gazelle.py

```diff
@@ -0,0 +1,499 @@
+    def schema(self):
+        """The schema of the plugin
+
+        Subclasses should extend this to implement more params
+        """
+        schema = {
+            'type': 'object',
+            'properties': {
+                'base_url': {'type': 'string'},
+                'username': {'type': 'string'},
+                'password': {'type': 'string'},
+                'max_pages': {'type': 'integer'},
+                'user_agent': {'type': 'string'},
+                'search': {'type': 'string'},
+            },
+            'required': ['username', 'password'],
+            'additionalProperties': False
+        }
+        # base_url is only required if the subclass doesn't set it
+        if not self.base_url:
+            schema['required'].append('base_url')
+        return schema
+
+    def _key(self, key):
+        """Gets the API key name from the entered key"""
+        if key in self.aliases:
+            return self.aliases[key]
+        return key
+
+    def _opts(self, key):
+        """Gets the options for the specified key"""
+        return self.params[self._key(key)]
+
+    def _getval(self, key, val):
+        """Gets the value for the specified key based on a config option"""
+        opts = self._opts(key)
+        if isinstance(opts, dict):
+            # Translate the input value to the API value
+            # The str cast converts bools to 'True'/'False' for use as keys
+            # This allows for options that have True/False/Other values
+            return opts[str(val)]
+        elif isinstance(val, list):
+            # Fix yaml parser making a list out of a string
+            return ",".join(val)
+        return val
+
+    def params_from_config(self, config):
+        """Filter params and map config values -> api values"""
+        ret = {}
+        for k, v in config.items():
+            key = self._key(k)
+            if key in self.params:
+                ret[key] = self._getval(k, v)
+        return ret
+
+    def setup(self, task, config):
+        """Set up a session and log in"""
+        self._session = task.requests
+        base_url = config.get('base_url', "").rstrip("/")
+        if base_url:
+            if self.base_url and self.base_url != base_url:
+                log.warning("Using plugin designed for %s on %s - "
+                            "things may break", self.base_url, base_url)
+            self.base_url = base_url
+
+        if not self.base_url:
+            raise PluginError("No 'base_url' configured")
+
+        # Any more than 5 pages is probably way too loose of a search
+        self.max_pages = config.get('max_pages', 5)
+
+        # The consistent request limiting rule seems to be:
+        # "Refrain from making more than five (5) requests every ten (10) seconds"
+        self._session.add_domain_limiter(TokenBucketLimiter(self.base_url, 2, '2 seconds'))
+
+        # Custom user agent
+        user_agent = config.get('user_agent', "Flexget [Gazelle plugin]")
```

Comment:
Is this necessary? We already set a user_agent in `task.requests` (https://github.com/Flexget/Flexget/blob/develop/flexget/utils/requests.py#L184)